src/Persistent_cohomology/include/gudhi/Persistent_cohomology/Field_Zp.h

```diff
@@ -34,15 +35,34 @@ class Field_Zp {
 
   void init(int charac) {
     assert(charac > 0);  // division by zero + non negative values
+
     Prime = charac;
+
+    // Check that the provided prime is less than the maximum allowed as int and calculation below : 46337 ; i.e (max_prime-1)**2 <= INT_MAX
+    if(Prime > 46337)
+        throw std::invalid_argument("Maximum homology_coeff_field allowed value is 46337");
+
+    // Check for primality
+    if ((Prime == 0) || (Prime == 1) || ((Prime > 3) && ((Prime % 2 == 0) || (Prime % 3 == 0))))
```

Comment:
`Prime <= 1` may suffice here, I think you will catch the others later.

Reply:
That was to avoid moving forward in calculation for "trivial" non-prime numbers but ok.